logstash-core/lib/logstash/codecs/base.rb

```diff
@@ -11,7 +11,7 @@ module LogStash::Codecs; class Base < LogStash::Plugin
 
   def initialize(params={})
     super
-    config_init(@params)
+    config_init(@allow_env, @params)
```

Comment:
on second thought, I'll leave it in so that `config_init` is responsible to take it out